src/Sulu/Component/SmartContent/ContentType.php

```diff
@@ -379,16 +380,21 @@ private function getProvider(PropertyInterface $property)
      * @param string $pageParameter
      *
      * @return int
+     *
+     * @throws PageOutOfBoundsException
      */
     private function getCurrentPage($pageParameter)
     {
-        if ($this->requestStack->getCurrentRequest() !== null) {
-            $page = $this->requestStack->getCurrentRequest()->get($pageParameter, 1);
-        } else {
-            $page = 1;
+        if ($this->requestStack->getCurrentRequest() === null) {
+            return 1;
+        }
+
+        $page = $this->requestStack->getCurrentRequest()->get($pageParameter, 1);
+        if ($page < 1 || $page > PHP_INT_MAX) {
```

Comment:
Hm... but we could throw the same exception where the items are loaded, in case `$page > 0 && count($pages) < 1`?